portality/bll/doaj.py

```diff
@@ -0,0 +1,620 @@
+        # that journal
+        if application.current_journal is not None:
+            cj = models.Journal.pull(application.current_journal)
+            if cj is not None:
+                # carry the id and the created date
+                journal.set_id(cj.id)
+                journal.set_created(cj.created_date)
+
+                # bring forward any notes from the old journal record
+                old_notes = cj.notes
+                for note in old_notes:
+                    journal.add_note(note.get("note"), note.get("date"))
+
+                # bring forward any related applications
+                related = cj.related_applications
+                for r in related:
+                    journal.add_related_application(r.get("application_id"), r.get("date_accepted"), r.get("status"))
+
+                # ignore any previously set bulk_upload reference
+
+                # carry over any properties that are not already set from the application
+                # * contact
+                # * editor & editor_group (together or not at all)
+                # * owner
+                if len(journal.contacts()) == 0:
+                    old_contacts = cj.contacts()
+                    for contact in old_contacts:
+                        journal.add_contact(contact.get("name"), contact.get("email"))
+                if journal.editor is None and journal.editor_group is None:
+                    journal.set_editor(cj.editor)
+                    journal.set_editor_group(cj.editor_group)
+                if journal.owner is None:
+                    journal.set_owner(cj.owner)
+
+        if app.logger.isEnabledFor("debug"): app.logger.debug("Completing application_2_journal")
+
+        return journal
+
+    def journal_2_application(self, journal, account=None, keep_editors=False):
+        """
+        Function to convert a given journal into an application object.
+
+        Provide the journal, and it will be converted
+        in-memory to the application object (currently a Suggestion).  The new application
+        WILL NOT be saved by this method.
+
+        If an account is provided, this will validate that the account holder is
+        allowed to make this conversion
+
+        :param journal: a journal to convert
+        :param account: an account doing the action - optional, if specified the application will only be created if the account is allowed to
+        :return: Suggestion object
+        """
+
+        # first validate the incoming arguments to ensure that we've got the right thing
+        argvalidate("journal_2_application", [
+            {"arg": journal, "instance" : models.Journal, "allow_none" : False, "arg_name" : "journal"},
+            {"arg" : account, "instance" : models.Account, "arg_name" : "account"}
+        ], exceptions.ArgumentException)
+
+        if app.logger.isEnabledFor("debug"): app.logger.debug("Entering journal_2_application")
```

Comment:
I think it's to do with the amount of effort you have to go to each time.  So, only bother creating a big string object in memory and passing it into the logging stack if it's worth it.  The isEnabledFor is quick and less memory intensitve, basically.  It's a small thing, but if we litter the code with more debug statements it could save some cycles in the long run.